src/GovUk.Education.ExploreEducationStatistics.Common/Extensions/ReflectionExtensions.cs

```diff
@@ -58,12 +58,7 @@ public static bool TryBoxToResult(this object value, Type targetType, out object
                         break;
 
                     case "Task":
-                        var task = typeof(Task).GetMethod("FromResult")
-                            ?.MakeGenericMethod(boxType.GenericTypeArguments)
-                            .Invoke(null, new[] { boxedValue });
-
-                        boxedValue = task;
-                        break;
+                        throw new ArgumentException("Should never need to box a Task");
```

Comment:
You mean remove the exception? i.e. no case statement for Tasks? I think maybe I'm just being overly cautious by throwing this exception. I think I'd keep it for now - but happy to go along with what others think.

I think it's right that the box/unbox things shouldn't deal with Tasks. My intuition now is that it's dangerous to let them do so - I think you'd agree with that but just saying to make sure we're on the same page.

Reply:
I'd still prefer to throw an exception because I think it prevents a lot of potential mischief - see comment below - but happy to change it if others want.

Reply:
Again, would prefer that we change this. The code isn't unsafe as we're not calling any synchronous methods, so the method should do what it says on the tin i.e. box it up.

Reply:
Restored original code and restored associated `TryBoxToResult` unit tests.